config/schedule.rb

```diff
@@ -37,6 +37,10 @@
   rake 'earthworks:clear_rack_attack_cache'
 end
 
+every 1.day, at: '3:04 am' do
+  rake 'rake earthworks:prune_old_guest_user_data[12]'
```

Comment:
I'm wondering if even 3 months would be ok? maybe even less?

We are getting around 1.5 million every 3 months :(

Reply:
Yeah, totally.  I think, as mentioned in the issue, it might be a good idea to do something w/ a crawler detector to reduce the number of guest sessions, but I think that can be handled separately.

One issue I guess that does raise, is that this isn't going to be great at killing millions of database rows at once (and will be best to keep things manageable once they are at a manageable level).  This might benefit from operating on a table that is somewhat w/i a reasonable size. 

I'm not sure where it's gotten today, but maybe we should take a look at running the manual user pruning tasks we ran before to get things down before putting this in place?

Reply:
I think thats a good idea. FWIW, I ran this on prod early morning and with 12 months it wasn't outrageously slow and completed withing 10 seconds.